File: 2656-maximum-sum-with-exactly-k-elements/2656-maximum-sum-with-exactly-k-elements.java
class Solution {
    public int maximizeSum(int[] nums, int k) {
        int x = -1;
        for (int num : nums) {
            if (x < num) {
                x = num;
            }
        }
        return k * x + (k - 1) * k / 2;
    }
}

File: 2656-maximum-sum-with-exactly-k-elements/2656-maximum-sum-with-exactly-k-elements.rs
impl Solution {
    pub fn maximize_sum(nums: Vec<i32>, k: i32) -> i32 {
    let mut largest = -1;

    for &num in nums.iter() {
        if largest < num {
            largest = num;
        }
    }

    let mut result = 0;

    for _ in 0..k {
        result += largest;
        largest += 1;
    }

    return result;
}

}